gammapy/estimators/points/lightcurve.py

```diff
@@ -151,6 +162,24 @@ def run(self, datasets):
 
             valid_intervals.append([t_min, t_max])
 
+            if self.stack_over_time_interval:
+                name = f"timebin_{t_min.mjd:.0f}_{t_max.mjd:.0f}"
+                dataset_reduced = datasets_to_fit.stack_reduce(name=name)
+                models = datasets.models.copy()
+                for dataset in datasets:
+                    models.reassign(dataset.name, dataset_reduced.name)
```

Comment:
If the datasets had background models they are already applied by stack reduce so you cannot reassign them to the reduced dataset you would need to drop the existing background models and create a new one. Regarding the models that applied to a specific dataset without being a background model I think that's fine.